typescript/src/schema/naming.ts

```diff
@@ -32,9 +42,13 @@ export class Naming {
     // common prefix must either end with `.`, or be equal to at least one of
     // the packages' prefix
     if (!prefix.endsWith('.') && !protoPackages.some(pkg => pkg === prefix)) {
-      throw new Error('Protos provided have different proto packages.');
-    }
-    const rootPackage = prefix.replace(/\.$/, '');
+      if (mainServiceName) {
+        rootPackage = this.checkServiceInPackage(
+          protoPackages,
+          mainServiceName
+        );
+      } else throw new Error('Protos provided have different proto packages.');
+    } else rootPackage = prefix.replace(/\.$/, '');
```

Comment:
Let's get rid of nested conditions. This can be rewritten as

```ts
rootPackage = '';
if (mainServiceName) {
  rootPackage = this.checkServiceInPackage(.....);
}
if (rootPackage === '') {
  rootPackage = prefix.replace(/\.$/, '');
}
if (rootPackage === '') {
  throw new Error(.....);
}
```

so we never go more than one level deep in `if`, and even have no `else` at all.